dds_web/user.py

```diff
@@ -22,29 +23,48 @@ def login():
     if request.method == "GET":
         if session.get("is_admin"):
             return redirect(url_for("admin.admin_page"))
-        elif session.get("current_user") and session.get("usid"):
+        elif session.get("current_user"):
             return redirect(url_for("user.user_page", loginname=session["current_user"]))
         else:
             return render_template("user/login.html", next=request.args.get("next"))
 
+    # Login form procedure
     if request.method == "POST":
+        # Get username and password from form
         username = request.form.get("username")
         password = request.form.get("password")
-        credentials_validated, is_facility, message, user_info = validate_user_credentials(
-            username, password
-        )
-        if not credentials_validated:
+
+        # Authenticate user
+        try:
+            user_ok = dds_auth.verify_user_pass(username=username, password=password)
+        except sqlalchemy.exc.SQLAlchemyError as sqlerr:
+            # TODO (ina): Create custom error page
+            app.logger.exception(sqlerr)
+            return str(sqlerr), 500
+        except exceptions.AuthenticationError as autherr:
+            app.logger.exception(autherr)
             return render_template(
-                "user/login.html", next=request.form.get("next"), login_error_message=message
+                "user/login.html", next=request.form.get("next"), login_error_message=str(autherr)
             )
-        session["current_user"] = user_info["username"]
-        session["current_user_id"] = user_info["id"]
-        session["is_admin"] = user_info.get("admin", False)
-        session["is_facility"] = is_facility
-        session["facility_name"] = user_info.get("facility_name")
-        session["facility_id"] = user_info.get("facility_id")
 
-        # temp admin fix
+        # User authentication not passed
+        if not user_ok:
```

Comment:
I think this is the same thing as in the other user.py... What is the difference between  **dds_web/api/user.py** and **dds_web/user.py** ?

Reply:
They are very similar. The difference is that `dds_web/user.py` uses pure flask, while `dds_web/api/user.py` uses Flask Restful (-> Resources), so `def login()` accepts both `GET` and `POST` while `AuthenticateUser(flask_restful.Resource)` atm just accepts get.

Also that they return different things - `dds_web/user.py` functions return templates and redirects to different pages in the web interface, and `dds_web/api/user.py` returns json to the CLI.